src/components/ui/OakTextInput/OakTextInput.tsx

```diff
@@ -0,0 +1,196 @@
+    box-shadow: ${(props) =>
+      props.$focusRingDropShadows
+        .map((dropShadow) => parseDropShadow(dropShadow))
+        .join(",")};
+  }
+
+  background: ${(props) => parseColor(props.$background)};
+
+  @media (hover: hover) {
+    &:hover:not(:focus-within, :has(input:read-only)) {
+      background: ${(props) => parseColor(props.$hoverBackground)};
+    }
+  }
+
+  &:has(input:read-only) {
+    border-color: ${(props) => parseColor(props.$readOnlyBorderColor)};
+    color: ${(props) => parseColor(props.$readOnlyColor)};
+  }
+
+  &:has(input:disabled) {
+    background: ${(props) => parseColor(props.$disabledBackgroundColor)};
+    color: ${(props) => parseColor(props.$disabledColor)};
+  }
+`;
+
+/**
+ * Default input which can be extended to create specialised inputs.
+ */
+export const OakTextInput = ({
+  type = "text",
+  $borderColor = "border-primary",
+  $readOnlyBorderColor = "border-neutral",
+  $focusRingDropShadows = [
+    "drop-shadow-centered-yellow",
+    "drop-shadow-centered-grey",
+  ],
+  $background = "bg-primary",
+  $hoverBackground = "bg-neutral",
+  $disabledBackgroundColor = "bg-neutral",
+  $color = "text-primary",
+  $disabledColor = "text-disabled",
+  $readOnlyColor = "text-subdued",
+  validity,
+  $iconColor = "icon-inverted",
+  $validBorderColor = "border-success",
+  $invalidBorderColor = "border-error",
+  $validIconColor = "icon-success",
+  $invalidIconColor = "border-error",
+  startEnhancerIconName,
+  endEnhancerIconName,
+  ...props
+}: OakTextInputProps) => {
+  let borderColor: OakCombinedColorToken;
+  let iconColor: OakCombinedColorToken;
+  let readOnlyBorderColor: OakCombinedColorToken;
+
+  switch (validity) {
+    case "valid":
+      borderColor = $validBorderColor;
+      iconColor = $validIconColor;
+      readOnlyBorderColor = $validBorderColor;
+      break;
+    case "invalid":
+      borderColor = $invalidBorderColor;
+      iconColor = $invalidIconColor;
+      readOnlyBorderColor = $invalidBorderColor;
+      break;
+    default:
+      borderColor = $borderColor;
+      iconColor = $iconColor;
+      readOnlyBorderColor = $readOnlyBorderColor;
+      break;
+  }
+
+  return (
+    <StyledTextInputWrapper
+      $width="fit-content"
+      $height="fit-content"
+      $borderStyle="solid"
+      $borderRadius="border-radius-s"
+      $ba="border-solid-m"
+      $borderColor={borderColor}
+      $focusRingDropShadows={$focusRingDropShadows}
+      $background={$background}
+      $hoverBackground={$hoverBackground}
+      $disabledBackgroundColor={$disabledBackgroundColor}
+      $readOnlyBorderColor={readOnlyBorderColor}
+      $disabledColor={$disabledColor}
+      $readOnlyColor={$readOnlyColor}
+      $color={$color}
+      $alignItems="center"
+      $position="relative"
+      $gap="space-between-s"
+      $ph="inner-padding-l"
+      onClick={(event) => {
+        event.currentTarget.querySelector("input")?.focus();
```

Comment:
Should we pass this event up as a callback ?

Reply:
🤔 thinking about this more I wonder if the more appropriate thing to do here is to also trigger `click` on the input.

The intent was for the div to proxy the input. If I bound an `onClick` to `OakInput` and `currentTarget` was the wrapper `div` I wouldn't have immediate access to the value of the `input` which might be unexpected.

As it stands if I pass `onClick` to `OakInput` it will be bound to the underlying `<input />`  but won't be called if I click the padding of the `div`.

```suggestion
        const domInput = event.currentTarget.querySelector("input");
        input?.focus();
        input?.click();
```